### ISSUE TITLE
Firestore: Internal error when fetching all the documents in a collection.

### ISSUE THREAD

**Issue by RubenEsposito** [firebase/firebase-ios-sdk#1731]
#### Describe your environment:

  * Xcode version: 9.4.1
  * Firebase SDK version: 4.3.0
  * Firebase Component: Firestore
  * Component version: 0.8.0

#### Describe the problem:

We tried to use the example iOS code to fetch all the documents on our Firestore collection from:

https://firebase.google.com/docs/firestore/query-data/get-data

It worked on the first run, but the next time we tried we received an internal error from Firestore:

```
*** Assertion failure in -[FSTLevelDBRemoteDocumentCache decodedMaybeDocument:withKey:], third_party/firebase/ios/Source/Firestore/Source/Local/FSTLevelDBRemoteDocumentCache.mm:147

*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'FIRESTORE INTERNAL ASSERTION FAILED: Read document has key (<FSTDocumentKey: AddressLatLngDictionary/A21 Flavour Studio, Teurastamo, Työpajankatu 2, 00580 Helsinki>) instead of expected key (<FSTDocumentKey: AddressLatLngDictionary/A21 Flavour Studio, Teurastamo, Työpajankatu 2, 00580 Helsink>).'
*** First throw call stack:
(0x184a66d8c 0x183c205ec 0x184a66bf8 0x185456fa0 0x1016cc8ac 0x1016cc2a8 0x1016cd120 0x1016ccf10 0x1016d20ec 0x1016b7964 0x1016afbc0 0x1016b14d8 0x106cc11dc 0x106cc119c 0x106ccfdfc 0x106cc46ac 0x106cd0d54 0x106cd7e38 0x18468be70 0x18468bb08)
libc++abi.dylib: terminating with uncaught exception of type NSException
```
It fails in the first field stating that the keys don't match (in one it is Helsinki and in the other Helsink).

#### Steps to reproduce:

We are using the plain code from the Firebase website, only changing the collection name. Just run the code pasted behind two times with an existing collection instead of "cities". In our case the documents fields contain UTF-8 characters and special characters: /,.

#### Relevant Code:

```
db.collection("cities").getDocuments() { (querySnapshot, err) in
    if let err = err {
        print("Error getting documents: \(err)")
    } else {
        for document in querySnapshot!.documents {
            print("\(document.documentID) => \(document.data())")
        }
    }
}
```

**Comment by wilhuff** [comment]
This is a dup of #445, the fix for which was released as part of Firebase 4.8.0 in December 2017.

You really should upgrade your SDK :-).

**Comment by RubenEsposito** [comment]
Thank you @wilhuff ! :)
